internal/bminventory/inventory.go

```diff
@@ -1941,8 +1958,25 @@ func (b *bareMetalInventory) updateNonDhcpNetworkParams(updates map[string]inter
 				log.WithError(err).Warnf("Verify dual-stack machine networks")
 				return common.NewApiError(http.StatusBadRequest, err)
 			}
+			secondaryMachineNetworkCidr, err = network.GetSecondaryMachineCidr(cluster)
+			if err != nil {
+				return common.NewApiError(http.StatusBadRequest, err)
+			}
+
+			if err = network.VerifyVips(cluster.Hosts, primaryMachineNetworkCidr, string(apiVips[0].IP), string(ingressVips[0].IP), false, log); err != nil {
+				log.WithError(err).Warnf("Verify VIPs")
+				return common.NewApiError(http.StatusBadRequest, err)
+			}
+
+			if len(ingressVips) == 2 && len(apiVips) == 2 { // in case there's a second set of VIPs
+				if err = network.VerifyVips(cluster.Hosts, secondaryMachineNetworkCidr, string(apiVips[1].IP), string(ingressVips[1].IP), false, log); err != nil {
+					log.WithError(err).Warnf("Verify VIPs")
+					return common.NewApiError(http.StatusBadRequest, err)
+				}
+			}
+
 		} else {
-			primaryMachineNetworkCidr, err = network.CalculateMachineNetworkCIDR(apiVip, ingressVip, cluster.Hosts, matchRequired)
+			primaryMachineNetworkCidr, err = network.CalculateMachineNetworkCIDR(string(apiVips[0].IP), string(ingressVips[0].IP), cluster.Hosts, matchRequired)
```

Comment:
Feel free to ignore this comment, but https://github.com/openshift/assisted-service/pull/4575/files will add a getter function to the VIPs that would allow you to write `GetApiVipById(cluster, 0) instead of `string(apiVips[0].IP)`